src/components/PopupWithForm.js

```diff
@@ -1,51 +1,52 @@
-import Popup from './Popup';
+import Popup from "./Popup";
 export default class PopupWithForm extends Popup {
-    constructor({ popupSelector, handleFormSubmit, loadingText }) {
-        super({ popupSelector });
-        this._popupForm = this._popupElement.querySelector('.modal__form');
-        this._inputList = this._popupElement.querySelectorAll('.modal__input');
-        this._handleFormSubmit = handleFormSubmit;
-        this._submitButton = this._popupForm.querySelector(".modal__button[type='submit']");
-        this._submitButtonText = this._submitButton.textContent;
-        this._loadingText = loadingText;
-    }
+  constructor({ popupSelector, handleFormSubmit, loadingText }) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupForm.querySelector(
+      ".modal__button[type='submit']"
+    );
+    this._submitButtonText = this._submitButton.textContent;
+    this._loadingText = loadingText;
+  }
 
-    setInputValues(data) {
-        this._inputList.forEach((input) => {
-            // here you insert the `value` by the `name` of the input
-            input.value = data[input.name];
-        });
-    }
+  setInputValues(data) {
+    this._inputList.forEach((input) => {
+      input.value = data[input.name];
+    });
+  }
 
-    close() {
-        this._popupForm.reset();
-        super.close();
-    }
-    _getInputValues() {
-        this._newData = {};
-        this._inputList.forEach((inputElement) => {
-            this._newData[inputElement.name] = inputElement.value;
-        });
-        return this._newData;
-    }
-    setSubmitAction(action) {
-        this._handleFormSubmit = action;
-    }
-    renderLoading(isLoading) {
-        if (isLoading) {
-            this._submitButton.textContent = this._loadingText;
-        } else {
-            this._submitButton.textContent = this._submitButtonText;
-        }
+  close() {
+    this._popupForm.reset();
+    super.close();
+  }
+  _getInputValues() {
+    this._newData = {};
+    this._inputList.forEach((inputElement) => {
+      this._newData[inputElement.name] = inputElement.value;
+    });
+    return this._newData;
+  }
+  setSubmitAction(action) {
```

Comment:
You should create a different class `PopupWithConfirmation` for handling the popup with delete confirmation